pkg/controller/installation/validation.go

```diff
@@ -291,8 +292,16 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			}
 		}
 
-		if bpfDataplane && instance.Spec.CalicoNetwork.NodeAddressAutodetectionV4 == nil {
-			return fmt.Errorf("spec.calicoNetwork.nodeAddressAutodetectionV4 is required for the BPF dataplane")
+		if bpfDataplane {
+			if bpfIPv6Enabled {
```

Comment:
Is dual stack supported with bpf? If not then I think we should have a check for that.
I think I'd also suggest moving these validations to under the `if v4pool != nil {` and `if v6pool != nil {` blocks.

Reply:
Dual stack is not supported. Added a check for that.